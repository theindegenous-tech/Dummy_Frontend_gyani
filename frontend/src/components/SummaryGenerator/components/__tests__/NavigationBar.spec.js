import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Navigation from '../NavigationBar'

describe('Navigation Bar', () => {
  const mockFn = jest.fn()

  it('Should render correctly', () => {
    const tree = renderer
      .create(
        <Navigation
          page={1}
          pages={10}
          scale={1}
          defaultScale={1}
          maxScale={3}
          minScale={1}
          rotationAngle={0}
          handleNextClick={mockFn}
          handlePrevClick={mockFn}
          handleZoomIn={mockFn}
          handleResetZoom={mockFn}
          handleZoomOut={mockFn}
          handleRotateLeft={mockFn}
          handleResetRotation={mockFn}
          handleRotateRight={mockFn}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should hide control on passing props', () => {
    const loader = shallow(
      <Navigation
        page={1}
        pages={10}
        scale={1}
        defaultScale={1}
        maxScale={3}
        minScale={1}
        rotationAngle={0}
        hideZoom
        hideRotation
        handleNextClick={mockFn}
        handlePrevClick={mockFn}
        handleZoomIn={mockFn}
        handleResetZoom={mockFn}
        handleZoomOut={mockFn}
        handleRotateLeft={mockFn}
        handleResetRotation={mockFn}
        handleRotateRight={mockFn}
      />
    )
    expect(loader.text()).toEqual(
      '<PreviousPageButton /><PageIndicator /><NextPageButton />'
    )
  })
})
