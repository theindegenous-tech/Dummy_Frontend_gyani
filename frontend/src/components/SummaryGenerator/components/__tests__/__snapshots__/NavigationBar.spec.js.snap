// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar Should render correctly 1`] = `
<div
  className="columns is-gapless box is-mobile has-text-white has-background-black py-4"
>
  <div
    className="column is-flex is-flex-direction-row is-4 buttons are-small has-addons"
  >
    <button
      className="button is-black my-0 mx-3"
      disabled={true}
    >
      <span
        className="icon is-small"
      >
        <i
          className="material-icons"
        >
          zoom_out
        </i>
      </span>
    </button>
    <button
      className="button is-black my-0 mx-3"
      disabled={true}
    >
      <span
        className="icon is-small"
      >
        <i
          className="material-icons"
        >
          refresh
        </i>
      </span>
    </button>
    <button
      className="button is-black my-0 mx-3"
      onClick={[MockFunction]}
    >
      <span
        className="icon is-small"
      >
        <i
          className="material-icons"
        >
          zoom_in
        </i>
      </span>
    </button>
  </div>
  <div
    className="column is-flex is-flex-direction-row is-4 buttons are-small has-addons is-centered"
  >
    <button
      className="button is-black my-0 mx-3"
      disabled={true}
    >
      <span
        className="icon is-small"
      >
        <i
          className="material-icons"
        >
          keyboard_arrow_left
        </i>
      </span>
    </button>
    <span
      className="is-size-7 has-text-centered my-0 mx-3"
    >
      Page 1 / 10
    </span>
    <button
      className="button is-black my-0 mx-3"
      onClick={[MockFunction]}
    >
      <span
        className="icon is-small"
      >
        <i
          className="material-icons"
        >
          keyboard_arrow_right
        </i>
      </span>
    </button>
  </div>
  <div
    className="column is-flex is-flex-direction-row is-4 buttons are-small has-addons is-right"
  >
    <button
      className="button is-black my-0 mx-3"
      onClick={[MockFunction]}
    >
      <span
        className="icon is-small"
      >
        <i
          className="material-icons"
        >
          rotate_left
        </i>
      </span>
    </button>
    <button
      className="button is-black my-0 mx-3"
      disabled={true}
    >
      <span
        className="icon is-small"
      >
        <i
          className="material-icons"
        >
          refresh
        </i>
      </span>
    </button>
    <button
      className="button is-black my-0 mx-3"
      onClick={[MockFunction]}
    >
      <span
        className="icon is-small"
      >
        <i
          className="material-icons"
        >
          rotate_right
        </i>
      </span>
    </button>
  </div>
</div>
`;
